tool_quality_checks/correct_import_file.py

```diff
@@ -0,0 +1,244 @@
+    out_feat.SetGeometry(polygon)
+
+    for key, value in content.items():
+        out_feat.SetField(key, value)
+
+    out_lyr.CreateFeature(out_feat)
+
+
+def fix_geometry(geometry):
+    geom_name = geometry.GetGeometryName()
+    # check pointcount if linestring
+    if "LINESTRING" in geom_name:
+        if geometry.GetPointCount() == 1:
+            logger.warning("Geometry point count of linestring = 1")
+            return geometry, False
+        else:
+            pass
+
+    # solve self intersections
+    if "POLYGON" in geom_name:
+        try:
+            geometry = geometry.Buffer(0)
+        except Exception as e:
+            # RuntimeError: IllegalArgumentException: Points of LinearRing do not form a closed linestring
+            logger.warning(e)
+            return geometry, False
+
+    # We think this part is not used or creates an error. So just leave it out for a while
+    # # check slivers
+    # if not geometry.IsValid():
+    #     perimeter = geometry.Boundary().Length()
+    #     area = geometry.GetArea()
+
+    #     sliver = float(perimeter / area)
+
+    #     if sliver < 1:
+    #         wkt = geometry.ExportToWkt()
+    #         geometry = ogr.CreateGeometryFromWkt(wkt)
+
+    return geometry, geometry.IsValid()
+
+
+def correct(in_layer, layer_name="", epsg=28992):
+    """
+        This function standardizes a vector layer:
+            1. Multipart to singleparts
+            2. 3D polygon to 2D polygon
+            3. Reprojection
+            4. Fix for self intersections
+            
+        INPUT: layer, layer_name
+        OUTPUT: out_datasource, layer_name
+      """
+
+    # retrieving lost features
+    lost_features = []
+    in_feature_count = in_layer.GetFeatureCount()
+
+    # Get inspatial reference and geometry from in shape
+    geom_type = in_layer.GetGeomType()
+    in_spatial_ref = in_layer.GetSpatialRef()
+    in_layer.ResetReading()
+
+    if layer_name != "":  # lower layer_name
+        layer_name = layer_name.lower()
+        logger.info("check - Name length")
+        if len(layer_name) + 10 > 64:
+            logger.warning("laagnaam te lang, 50 characters max.")
+            logger.warning("formatting naar 50 met deze naam: %s" % layer_name[:50])
+            layer_name = layer_name[:50]
+
+    # create datasource output
+    out_datasource = create_mem_ds()
+    spatial_ref_out = osr.SpatialReference()
+    spatial_ref_out.ImportFromEPSG(int(epsg))
+    out_layer = out_datasource.CreateLayer(layer_name, spatial_ref_out, geom_type)
+
+    # create memory datasource for
+    mem_datasource = create_mem_ds()
+    mem_layer = mem_datasource.CreateLayer(layer_name, in_spatial_ref, geom_type)
+
+    layer_defn = in_layer.GetLayerDefn()
+    for i in range(layer_defn.GetFieldCount()):
+        field_defn = layer_defn.GetFieldDefn(i)
+        mem_layer.CreateField(field_defn)
+
+    logger.info("check - Multipart to singlepart")
+    lost_feat = multipoly2poly(in_layer, mem_layer)
```

Comment:
`lost_feat`: uit die `multipoly2poly()` kwam meervoud `lost_features`. Zoiets lijkt me duidelijker:
```suggestion
    additional_lost_features = multipoly2poly(in_layer, mem_layer)
```

Reply:
Kan die `multipoly2poly()` nog duidelijker qua naam? Hier gebruik je 'em.